### PR TITLE
Adds z stream release notes

### PR DESCRIPTION
**Pull request by DhritiShikhar**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

Adds documentation about z stream release

## TODO:

[ ] Add Bugzilla information

### CODE REVIEW

File: REBASE.openshift.md [hunk 1]
```diff
@@ -408,3 +408,90 @@ git log v1.21.1..v1.21.2 --ancestry-path --reverse --no-merges
...
+git remote add openshift `https://github.com/openshift/kubernetes`
+```
+
+3. Fetch everything from the remote `openshift`
+
+```
+git fetch openshift
+```
+
+4. Checkout to release-4.8 branch
```
Comment (lmzuccarelli) [inline REBASE.openshift.md:432]: How about we put something like release-Y.X branch  (as an example  for 4.8 it would be release-4.8)

Comment (DhritiShikhar) [reply REBASE.openshift.md]: Addressed this comment.

File: REBASE.openshift.md [hunk 2]
```diff
@@ -408,3 +408,90 @@ git log v1.21.1..v1.21.2 --ancestry-path --reverse --no-merges
...
+
+```
+git checkout openshift/release-4.8
+```
+
+5. Merge the latest kubernetes version and 
+
+```
+git merge v1.24.4
```
[the remark this answers is not in the document]
Comment (DhritiShikhar) [reply REBASE.openshift.md]: I have added some information about this. If it's still not clear, we could modify it.

File: REBASE.openshift.md [hunk 3]
```diff
@@ -408,3 +408,90 @@ git log v1.21.1..v1.21.2 --ancestry-path --reverse --no-merges
...
+12. Create a commit with the message:
+
+```
+UPSTREAM: <drop>: make update
+```
+
+13. Finally, push all the changes
+
+```
+git push Dhriti eb673a488fa:refs/heads/rebase-1.21.4
```
[the remark this answers is not in the document]
Comment (DhritiShikhar) [reply REBASE.openshift.md]: yes. Addressed this comment.

File: REBASE.openshift.md [hunk 4]
```diff
@@ -408,3 +408,110 @@ git log v1.21.1..v1.21.2 --ancestry-path --reverse --no-merges
...
+```
+git merge <version>
+```
+Example:
+
+```
+git merge v1.21.4
+````
+
+6. Create a commit **with the conflicts** using the following command:
```
Comment (soltysh) [inline REBASE.openshift.md:453]: That's wrong, normally during `git merge` you'll get a merge commit, example: https://github.com/openshift/kubernetes/pull/916/commits/c0f1a8fb7e8b50ba7f4734d2d0124e38b7f2ddeb

File: REBASE.openshift.md [hunk 5]
```diff
@@ -408,3 +408,110 @@ git log v1.21.1..v1.21.2 --ancestry-path --reverse --no-merges
...
+
+```
+git commit -m "UPSTREAM: <drop>: manually resolve conflicts"
+```
+
+7. Resolve the conflicts manually. In most cases, pick the latest version of dependencies.
```
Comment (soltysh) [inline REBASE.openshift.md:459]: Resolve conflicts should be separate commit, like so: https://github.com/openshift/kubernetes/pull/916/commits/78acef811bafc2a96df867e2a060a11eb6c81225 with `UPSTREAM: <drop>: manually resolve conflicts` to show reviewers that this is the most critical part to look after.

File: REBASE.openshift.md [hunk 6]
```diff
@@ -408,3 +408,110 @@ git log v1.21.1..v1.21.2 --ancestry-path --reverse --no-merges
...
+
+8. If `go.mod` file is modified in the previous step, then:
+- In the go.mod file, Update all `github.com/openshift/*` packages' version to the release branch
+Example: `github.com/openshift/api v0.0.0-20210817132244-67c28690af52` is changed to `github.com/openshift/api release-4.8` for `4.8` rebase
+- Run `go mod tidy`
+
+If `go.mod` file is not modified, then skip this step.
+
+9. Run the script `hack/update-vendor.sh`
```
Comment (soltysh) [inline REBASE.openshift.md:468]: 8 and 9 should be squashed into a single step.

File: REBASE.openshift.md [hunk 7]
```diff
@@ -408,3 +408,110 @@ git log v1.21.1..v1.21.2 --ancestry-path --reverse --no-merges
...
+
+10. Create a commit with all the changes using the following command:
+
+```
+git commit -m "UPSTREAM: <drop>: hack/update-vendor.sh"
+```
+
+11. Update kubernetes version in `openshift-hack/images/hyperkube/Dockerfile.rhel`
+
+12. Run `podman run -it --rm -v $( pwd ):/go/k8s.io/kubernetes:Z --workdir=/go/k8s.io/kubernetes registry.ci.openshift.org/openshift/release:rhel-8-release-golang-1.16-openshift-4.8 make update OS_RUN_WITHOUT_DOCKER=yes
```
Comment (soltysh) [inline REBASE.openshift.md:478]: It's best to use pre-existing docs, point to https://github.com/openshift/kubernetes/blob/master/REBASE.openshift.md#updating-generated-files

File: REBASE.openshift.md (hunk 6 above)
Comment (soltysh) [reply REBASE.openshift.md:468]: You can also point to https://github.com/openshift/kubernetes/blob/master/REBASE.openshift.md#updating-dependencies

File: REBASE.openshift.md [hunk 8]
```diff
@@ -408,3 +408,110 @@ git log v1.21.1..v1.21.2 --ancestry-path --reverse --no-merges
...
+`
+
+13. Create a commit with all the changes using the following command:
+
+```
+git commit -m "UPSTREAM: <drop>: make update"
```
Comment (soltysh) [inline REBASE.openshift.md:484]: I want this to be similar to https://github.com/openshift/kubernetes/pull/916/commits/687a5bf8293aad12e1e93eac7bafe58b19ab9b51 so this should be `UPSTREAM: <drop>: hack/update-vendor.sh, make update and update image`

File: REBASE.openshift.md [hunk 9]
```diff
@@ -408,3 +408,110 @@ git log v1.21.1..v1.21.2 --ancestry-path --reverse --no-merges
...
+
+```
+git push <username> <current-branch>:<remote-branch>
+```
+
+### Pull Request Best Practices:
+
+1. The pull request will always be against a particular release. So, Ensure the pull request is against the specific release branch e.g. `release-4.9` branch of `openshift/kubernetes` repository.
+
+2. Put the changes from last release to current release in the pull request description:
```
Comment (soltysh) [inline REBASE.openshift.md:497]: This I mentioned should be changed to use CHANGELOG from upstream.

File: REBASE.openshift.md [hunk 10]
```diff
@@ -408,3 +408,110 @@ git log v1.21.1..v1.21.2 --ancestry-path --reverse --no-merges
...
+git log <old-release>..<new-release> --ancestry-path --reverse --no-merges --oneline
+```
+
+Example to get changes from `v1.21.1` to `v1.21.4` use the following command:
+
+```
+git log v1.21.1..v1.21.4 --ancestry-path --reverse --no-merges --oneline
+```
+
+### How to pull images from registry.ci.openshift.org
```
Comment (soltysh) [inline REBASE.openshift.md:509]: Below are irrelevant to the procedure and should not be here.

**Review by soltysh — CHANGES_REQUESTED** [review]

**Comment by tjungblu** [comment]
@DhritiShikhar are we going to get this in? can you please fix the remaining points Maciej mentioned?
